Author: Pino Toscano <pino@debian.org>
Description: Make QtWebEngine an optional requirement
 This makes it possible to build konqueror, albeith without an HTML component,
 also on architectures that do not have QtWebEngine.
Forwarded: no
Last-Update: 2020-05-02

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,8 @@ include(CMakePackageConfigHelpers)
 set(KONQUEROR_LIB_VERSION "5.0.97")
 set(KONQUEROR_VERSION "${KONQUEROR_LIB_VERSION}")
 
-find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS Core Widgets WebEngineWidgets)
+find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS Core Widgets)
+find_package(Qt5 ${QT_MIN_VERSION} COMPONENTS WebEngineWidgets)
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS Parts KCMUtils Archive Crash WindowSystem Init IconThemes DBusAddons)
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS KDELibs4Support)
 
@@ -60,7 +61,9 @@ add_subdirectory( client )
 if(BUILD_TESTING)
     add_subdirectory( autotests )
 endif()
-add_subdirectory( webenginepart )
+if (Qt5WebEngineWidgets_FOUND)
+    add_subdirectory( webenginepart )
+endif()
 
 add_subdirectory( pics )
 #add_subdirectory( sidebar )
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -7,8 +7,10 @@ include_directories( ${CMAKE_CURRENT_SOU
 
 ########### konqviewmgrtest ###############
 
-ecm_add_test(konqviewmgrtest.cpp
-    LINK_LIBRARIES kdeinit_konqueror Qt5::Core Qt5::Gui kwebenginepartlib Qt5::WebEngineWidgets Qt5::Test)
+if (Qt5WebEngineWidgets_FOUND)
+  ecm_add_test(konqviewmgrtest.cpp
+      LINK_LIBRARIES kdeinit_konqueror Qt5::Core Qt5::Gui kwebenginepartlib Qt5::WebEngineWidgets Qt5::Test)
+endif()
 
 ########### historymanagertest ###############
 
@@ -22,8 +24,10 @@ ecm_add_test(undomanagertest.cpp
 
 ########### konqhtmltest ###############
 
-ecm_add_test(konqhtmltest.cpp
-    LINK_LIBRARIES kdeinit_konqueror kwebenginepartlib Qt5::Core Qt5::Test)
+if (Qt5WebEngineWidgets_FOUND)
+  ecm_add_test(konqhtmltest.cpp
+      LINK_LIBRARIES kdeinit_konqueror kwebenginepartlib Qt5::Core Qt5::Test)
+endif()
 
 ########### konqviewtest ###############
 
